sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs

```diff
@@ -298,7 +299,21 @@ public BlobCollectionConverter(BlobsExtensionConfigProvider parent)
                 }
             }
 
-            public async Task<IEnumerable<T>> ConvertAsync(MultiBlobContext context, CancellationToken cancellationToken)
+            async Task<T[]> IAsyncConverter<MultiBlobContext, T[]>.ConvertAsync(MultiBlobContext context, CancellationToken cancellationToken)
+            {
+                // Query the blob container using the blob prefix (if specified)
+                // Note that we're explicitly using useFlatBlobListing=true to collapse
+                // sub directories.
+                string prefix = context.Prefix;
```

Comment:
Why are you duplicating all this code? You should instead just call the other implementation:

```csharp
async Task<T[]> IAsyncConverter<MultiBlobContext, T[]>.ConvertAsync(MultiBlobContext context, CancellationToken cancellationToken)
{
    IEnumerable<T> result = await ((IAsyncConverter<MultiBlobContext, IEnumerable<T>>)this).ConvertAsync(context, cancellationToken).ConfigureAwait(false);
    T[] convertedResult = result.ToArray();
    return convertedResult;
}
```

Reply:
updated